lib/Parser.js

```diff
@@ -1777,7 +1777,14 @@ class Parser extends Tapable {
 		const args = options.map(renameArgOrThis);
 		const wasTopLevel = this.scope.topLevelScope;
 		this.scope.topLevelScope = false;
-		this.inScope(params.filter((identifier, idx) => !args[idx]), () => {
+		const scopeParams = params.filter((identifier, idx) => !args[idx]);
+
+		// Add function name in scope for recursive calls
+		if (functionExpression.id) {
+			scopeParams.push(functionExpression.id.name);
```

Comment:
I think this need to be `unshift` instead to catch this case:

```js
(function someFn(arg = someFn) {
```

Reply:
Just wrote a new test case for this situation with the default parameter and it seems to work as expected. Could you detail in which case it would not work?

Reply:
ok great